en/03_Drawing_a_triangle/03_Drawing/01_Command_buffers.md

```diff
@@ -276,8 +276,29 @@ vkCmdBindPipeline(commandBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS, graphicsPipeli
 
 The second parameter specifies if the pipeline object is a graphics or compute
 pipeline. We've now told Vulkan which operations to execute in the graphics
-pipeline and which attachment to use in the fragment shader, so all that remains
-is telling it to draw the triangle:
+pipeline and which attachment to use in the fragment shader.
+
+As noted in the [fixed functions chapter](../02_Graphics_pipeline_basics/02_Fixed_functions.md#dynamic-state), 
+we did specify viewport and scissor state for this pipeline to be dynamic.
+So we need to set them in the command buffer before issuing our draw command:
+
+```c++
+VkViewport viewport{};
+viewport.x = 0.0f;
+viewport.y = 0.0f;
+viewport.width = (float) swapChainExtent.width;
+viewport.height = (float) swapChainExtent.height;
```

Comment:
prefer using `static_cast<float>(swapChainExtent)`?  (general C++ best practices)

Reply:
Good point. The rest of the tutorial also uses the C++ style casts. I have changed that.